src/debug_operations.ipp

```diff
@@ -53,11 +53,22 @@ bool TwoFourTree<K,C,A>::Node::validateRelationships() const {
 				std::cout << check.second->parent_->getString() << "\n";
 		}
 
+		int num_children = 0;
 		for (int i = 0; i < node->num_keys_+1; i++) {
 			if (node->children_[i]) {
 				allNodes.push_back(std::make_pair(node,node->children_[i].get()));
+				++num_children;
 			}
 		}
+		if (!node->isLeaf() && num_children != node->num_keys_ + 1) {
```

Comment:
Considered adding check for 0 children for leaf nodes... but we determine whether a node is a leaf or not by looking at whether it has the very first child.
Maybe the for loop before this should be extended and check for all possible children. However, knowing that I'm likely to switch to a forward_list instead of an array, I'll probably end up reverting it so this can stay as it is for now.

Reply:
Loop extended to check all children
0 child check for leaves does not make sense because the definition of a leaf depends on the first child